Realm/Realm/Linq/WhereClauseVisitor.cs

```diff
@@ -0,0 +1,329 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+using Newtonsoft.Json;
+using Realms.Schema;
+using LazyMethod = System.Lazy<System.Reflection.MethodInfo>;
+
+namespace Realms
+{
+    internal class RealmLinqExpression : Expression
+    {
+        public ExpressionNode ExpressionNode { get; private set; }
+
+        public static RealmLinqExpression Create(ExpressionNode exp)
+        {
+            return new RealmLinqExpression { ExpressionNode = exp };
+        }
+    }
+
+    internal class WhereClauseVisitor : ExpressionVisitor
+    {
+        private static class Methods
+        {
+            internal static LazyMethod Capture<T>(Expression<Action<T>> lambda)
+            {
+                return new LazyMethod(() =>
+                {
+                    var method = (lambda.Body as MethodCallExpression).Method;
+                    if (method.IsGenericMethod)
+                    {
+                        method = method.GetGenericMethodDefinition();
+                    }
+
+                    return method;
+                });
+            }
+
+            internal static class String
+            {
+                internal static readonly LazyMethod Contains = Capture<string>(s => s.Contains(string.Empty));
```

Comment:
Let's remove all the things that are not working or that are not being used, including comments. This will make the code easier to read for now